infra/aws/terraform/modules/kms/kms.tf

```
@@ -0,0 +1,30 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_kms_key" "main" {
+  description             = var.description
+  deletion_window_in_days = var.deletion_window_in_days
+  is_enabled              = true
+  multi_region            = true
+  enable_key_rotation     = var.enable_key_rotation
+  policy                  = var.policy
```

```suggestion
  policy                  = jsonencode(jsonencode(var.policy)
```

---

Do you mean `(jsonencode(var.policy)` ?

---

Ah, no, I meant this:
```suggestion
  policy                  = jsonencode(jsondecode(var.policy))
```

Round trip via a JSON decoder. I don't do this as often now that Terraform has validations, though.

---

Can we instead use `data "aws_iam_policy_document"`  to do the validation ?

---

Yep, sure!

---

Done.